src/main.rs

```diff
@@ -178,44 +178,34 @@ enum TableError {
     Pager(PagerError),
 }
 
-impl Table {
-    fn new<P>(filename: P) -> Result<Self, TableError>
-    where
-        P: AsRef<Path>,
-    {
-        Pager::new(filename)
-            .map_err(TableError::Pager)
-            .map(|pager| {
-                let num_rows: u32 = (pager.file_length / ROW_SIZE as u64) as u32;
-                Table { num_rows, pager }
-            })
-    }
+struct VirtualMachine<'a> {
```

Comment:
Just wanted something to represent the "virtual machine" referred to whenever discussing the execute and select execution functions.